File: frontend/constant/leave.ts
import { LeaveTypeEnum, LeaveStatusEnum } from "@/types/leave.types";

export const leaveTypeLabels: Record<LeaveTypeEnum, string> = {
  [LeaveTypeEnum.YEAR]: "İllik",
  [LeaveTypeEnum.ILLNESS]: "Xəstəlik",
  [LeaveTypeEnum.FREE]: "Ödənişsiz",
};

export const leaveStatusLabels: Record<LeaveStatusEnum, string> = {
  [LeaveStatusEnum.PENDING]: "Gözləyir",
  [LeaveStatusEnum.SUCCESS]: "Təsdiqləndi",
  [LeaveStatusEnum.REJECT]: "Rədd edildi",
};

export const leaveStatusStyles: Record<LeaveStatusEnum, string> = {
  [LeaveStatusEnum.PENDING]: "bg-amber-50 text-amber-600",
  [LeaveStatusEnum.SUCCESS]: "bg-emerald-50 text-emerald-600",
  [LeaveStatusEnum.REJECT]: "bg-red-50 text-red-600",
};

File: backend/app/models/leave.py
from enum import Enum
from tortoise import models, fields


class LeaveType(str, Enum):
    YEAR = "İllik"
    ILLNESS = "Xəstəlik"
    FREE = "Ödənişsiz"


class LeaveStatus(str, Enum):
    PENDING = "Gözləyir"
    SUCCESS = "Təsdiqləndi"
    REJECT = "Rədd edildi"


class Leave(models.Model):
    id = fields.UUIDField(pk=True)
    employee = fields.TextField()
    type = fields.CharEnumField(LeaveType, max_length=32)
    start_leave = fields.DateField()
    end_leave = fields.DateField()
    day = fields.IntField()
    status = fields.CharEnumField(
        LeaveStatus, max_length=32, default=LeaveStatus.PENDING
    )
